src/auth/auth.service.ts

```diff
@@ -53,17 +53,21 @@ export class AuthService {
     }
     const accessToken = await this.getAuthTokenAndValidate(keyId, secret);
     await this.setLastUsedToken(keyId);
-    return await this.usersService.getUserByUsername(accessToken.user.username);
+    return await this.usersService.getUserByUsername(
+      (
+        await accessToken.user
+      ).username,
+    );
   }
 
   async createTokenForUser(
     user: User,
     identifier: string,
     validUntil: TimestampMillis,
   ): Promise<AuthTokenWithSecretDto> {
-    user.authTokens = await this.getTokensByUser(user);
+    user.authTokens = this.getTokensByUser(user);
```

Comment:
Maybe a type hint would be useful here?

Reply:
I'm not declaring a variable here (the `User` class defines the type of the property), so I can't an explicit type.